packages/cli/test/unit/openapi/schema-model.unit.js

```diff
@@ -63,6 +69,140 @@ describe('schema to model', () => {
+          {
+            name: 'apis',
+            signature:
+              'apis?: {\n  apiKey?: string;\n  apiVersionNumber?: string;\n  ' +
+              'apiUrl?: string;\n  apiDocumentationUrl?: string;\n}[];',
+            decoration: "@property({name: 'apis'})",
+          },
+        ],
+        imports: [],
+        import: "import {DataSetList} from './data-set-list.model';",
+        kind: 'class',
+        declaration:
+          '{\n  total?: number;\n  apis?: {\n  apiKey?: string;\n  ' +
+          'apiVersionNumber?: string;\n  apiUrl?: string;\n  ' +
+          'apiDocumentationUrl?: string;\n}[];\n}',
+        signature: 'DataSetList',
+      },
+      {
+        description: 'AnonymousType_2',
+        name: 'AnonymousType_2',
+        className: 'AnonymousType2',
+        fileName: 'anonymous-type-2.model.ts',
+        properties: [],
+        imports: [],
+        import: "import {AnonymousType2} from './anonymous-type-2.model';",
+        declaration: 'string',
+        signature: 'AnonymousType2',
+      },
+      {
+        description: 'AnonymousType_3',
+        name: 'AnonymousType_3',
+        className: 'AnonymousType3',
+        fileName: 'anonymous-type-3.model.ts',
+        properties: [],
+        imports: [],
+        import: "import {AnonymousType3} from './anonymous-type-3.model';",
+        declaration: 'string',
+        signature: 'AnonymousType3',
+      },
+      {
+        description: 'AnonymousType_4',
+        name: 'AnonymousType_4',
+        className: 'AnonymousType4',
+        fileName: 'anonymous-type-4.model.ts',
+        properties: [],
+        imports: [],
+        import: "import {AnonymousType4} from './anonymous-type-4.model';",
+        declaration: 'string',
+        signature: 'AnonymousType4',
+      },
+      {
+        description: 'AnonymousType_5',
+        name: 'AnonymousType_5',
+        className: 'AnonymousType5',
+        fileName: 'anonymous-type-5.model.ts',
+        properties: [],
+        imports: [],
+        import: "import {AnonymousType5} from './anonymous-type-5.model';",
+        declaration: 'string',
+        signature: 'AnonymousType5',
+      },
+      {
+        description: 'AnonymousType_6',
+        name: 'AnonymousType_6',
+        className: 'AnonymousType6',
+        fileName: 'anonymous-type-6.model.ts',
+        properties: [],
+        imports: [],
+        import: "import {AnonymousType6} from './anonymous-type-6.model';",
+        declaration: 'string',
+        signature: 'AnonymousType6',
+      },
+      {
+        description: 'AnonymousType_7',
+        name: '{\n  \n}[]',
+        className: 'AnonymousType7',
+        fileName: 'anonymous-type-7.model.ts',
+        properties: [],
+        imports: [],
+        import: "import {AnonymousType7} from './anonymous-type-7.model';",
+        declaration: '{\n  \n}[]',
+        signature: 'AnonymousType7',
+        itemType: {
+          imports: [],
+          declaration: '{\n  \n}',
```

Comment:
IIUC, this is defining an empty object type that does not have any properties, is my understanding correct? What is the intended usage of such type? 

Assuming this type is representing the following response schema:

```yaml
              schema:
                type: array
                items:
                  type: object
                  additionalProperties:
                    type: object
```

According to JSON Schema (https://json-schema.org/understanding-json-schema/reference/object.html#properties), `additionalProperties: {type: 'object'}` means that the item objects can contain any additional properties holding values of object type. For example:

```ts
response = [
  {foo: {flag: true, name: 'first'}},
  {bar: {flag: false}}
];
```

I think you need to take `additionalProperties` into consideration when emitting TypeScript declaration:

```ts
{
  [additionalProperty: string]: object
}
```

This may be out of scope of this pull request, IDK.

Reply:
Fixed.